pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -364,36 +364,55 @@ func (m *ManagerImpl) isVersionCompatibleWithPlugin(versions []string) bool {
 	return false
 }
 
+func (m *ManagerImpl) getDevicesToReuse(podUID string) (map[string]sets.String, bool) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	devices, ok := m.devicesToReuse[podUID]
+	return devices, ok
+}
+
+func (m *ManagerImpl) seDevicesToReuse(podUID string, devices map[string]sets.String) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	m.devicesToReuse[podUID] = devices
+}
+
+func (m *ManagerImpl) clearDevicesToReuse(podUID string) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	delete(m.devicesToReuse, podUID)
+}
+
 // Allocate is the call that you can use to allocate a set of devices
 // from the registered device plugins.
 func (m *ManagerImpl) Allocate(pod *v1.Pod, container *v1.Container) error {
-	if _, ok := m.devicesToReuse[string(pod.UID)]; !ok {
-		m.devicesToReuse[string(pod.UID)] = make(map[string]sets.String)
+	if _, ok := m.getDevicesToReuse(string(pod.UID)); !ok {
```

Comment:
I think get and set should be handled in the same function.
just like this：

devicesToReuse := m.createDevicesToReuseIfNotExist(string(pod.UID));

func (m *ManagerImpl) createDevicesToReuseIfNotExist(podUID string) map[string]sets.String {
	m.mutex.Lock()
	defer m.mutex.Unlock()
	if _, ok := m.devicesToReuse[podUID]; !ok {
            m.devicesToReuse[podUID] = make(map[string]sets.String)
        }
	return m.devicesToReuse[podUID]
}

Reply:
Good idea. 
The set is only used for initialization.